<launch>
    <arg name="bag" default="false" />

    <group if="$(eval arg('bag') == false)">
        <!-- ////////////////////////////////LAUNCH THE CAMERA FIRST\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\ -->
        <!-- Configure the IP address of SceneScan/SP1 -->
        <arg name="device_address" value="160.69.69.5" />

        <!-- Configure the destination where the calibration will saved -->
        <arg name="calib_file" value="/tmp/nerian_calib.yaml" />

        <!-- Download the current camera calibration -->
        <node pkg="nerian_stereo" type="download_calibration.sh" args="$(arg device_address) $(arg calib_file)" name="download_calib" output="screen" />

        <!-- Configure SceneScan/SP1 operation mode to stereo matching -->
        <!-- <node pkg="nerian_stereo" type="config_device_stereo.sh" args="$(arg device_address)" name="config_device" output="screen" /> -->

        <!-- For debugging it is possible to configure a launch prefix -->
        <!-- <arg name="launchPrefix" value="gdb -ex 'catch throw' -ex run \-\-args" /> -->
        <!-- <arg name="launchPrefix" value="valgrind" /> -->
        <arg name="launchPrefix" value="" />

        <!-- Launch the stereo node -->
        <node pkg="nerian_stereo" type="nerian_stereo_node" name="nerian_stereo_node" output="screen" launch-prefix="$(arg launchPrefix)">
            <!-- possible values: none, mono8, rgb8, rgb32f -->
            <param name="point_cloud_intensity_channel" type="string" value="mono8" />

            <param name="ros_coordinate_system" type="bool" value="true" />
            <param name="ros_timestamps" type="bool" value="true" />

            <!-- Possible color coding schemes "rainbow" "red_blue" "none" -->
            <param name="color_code_disparity_map" type="string" value="none" />
            <param name="color_code_legend" type="bool" value="true" />

            <param name="use_tcp" type="bool" value="false" />

            <param name="remote_host" type="string" value="$(arg device_address)" />
            <param name="remote_port" type="string" value="7681" />

            <param name="calibration_file" type="string" value="$(arg calib_file)"/>
            <param name="frame" type="string" value="map" />

            <param name="delay_execution" type="double" value="2" />
            <param name="max_depth" type="double" value="-1" />
        </node>
    </group>

    <!-- ////////////////////////////////LAUNCH THE RANGEMAP NODE\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\ -->
    <node name="preprocessing_node" type="preprocessing_node" pkg="ugoe_gap_detection_ros" output="screen"/>

    <!-- ////////////////////////////////LAUNCH THE GAP DETECTOR NODE\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\ -->
    <node name="gap_detector" type="gap_detector.py" pkg="ugoe_gap_detection_ros" output="screen"/>

    <!-- ////////////////////////////////LAUNCH RVIZ\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\ -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find ugoe_gap_detection_ros)/rviz_config.rviz" />

    <!-- ////////////////////////////////PLAY THE ROSBAG IF bag:= true\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\ -->
    <arg name="bag_file" default="$(find ugoe_gap_detection_ros)/demo/demo.bag"/> <!-- Change path for bagfile-->

    <node name="player" pkg="rosbag" type="play" output="screen" args="-q -l -r 1.0 --clock $(arg bag_file)" if="$(arg bag)"/>
    
    <!-- ////////////////////////////////START DYNAMIC RECONFIGURE\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\ -->
    <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui" args="-s reconfigure"/>
</launch>



